scrape/forms_test.go

```diff
@@ -0,0 +1,73 @@
+package scrape
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"golang.org/x/net/html"
+)
+
+func Test_ParseForms(t *testing.T) {
+	tests := []struct {
+		html  string
+		forms []Form
+	}{
+		{`<html></html>`, nil},
+		{`<html><form></form></html>`, []Form{{Values: url.Values{}}}},
+		{
+			`<html><form action="a" method="m"><input name="n1" value="v1"></form></html>`,
+			[]Form{{Action: "a", Method: "m", Values: url.Values{"n1": {"v1"}}}},
+		},
+		{
+			`<html>
+			  <form action="a1" method="m1"><input name="n1" value="v1"></form>
+			  <form action="a2" method="m2"><input name="n2" value="v2"></form>
+			</html>`,
+			[]Form{
+				{Action: "a1", Method: "m1", Values: url.Values{"n1": {"v1"}}},
+				{Action: "a2", Method: "m2", Values: url.Values{"n2": {"v2"}}},
+			},
+		},
+	}
+
+	for _, tt := range tests {
```

Comment:
Please use:

```
for _, tt := range tests {
   t.Run(tt.description, func(t *testing.T){
       ....
   })
}
```

This makes the logs clear which test case fails (should it ever fail).

Reply:
done